tx-submit: Better handling of hex encoded data

erikd opened the pull request:
This should allow the submission of both raw binary and hex encoded
transactions.

Closes: https://github.com/input-output-hk/cardano-explorer/issues/235

erikd reviewed (commented):
Should really use `Content-Transfer-Encoding` here and reject anything that does not match the encoding.

dcoutts reviewed (dismissed):
I suggest we do no content detection but rely only on the content type HTTP header.

I suggest we accept only `application/cbor` content type. To be helpful we can have the endpoint fail with any other content type, and include in the human readable error message part that it must be `application/cbor` which is of course raw binary.

erikd commented:
`application/cbor` makes sense. It should also be able to detect if the provided input is hex encoded and provide a decent error message in that case.

dcoutts commented:
Detection for the purpose of error messages is fine.

erikd committed:
tx-submit: Better error message if provided hex data

If the transaction to be submitted fails to decode, specicalize the error
message if the data is hex encoded.

Closes: https://github.com/input-output-hk/cardano-explorer/issues/235

erikd committed:
tx-submit: Use application/cbor as content-type

dcoutts reviewed (approved):
LGTM.

erikd committed:
Nix voodoo

erikd committed:
tx-submit: Document tx-submit-webapi and update Changelog
